cmd/algorand-indexer/daemon.go

```diff
@@ -102,6 +102,17 @@ var daemonCmd = &cobra.Command{
 			}()
 		} else {
 			logger.Info("No block importer configured.")
+			waitForDBAvailable(db)
```

Comment:
Should this block of code go into the if? I think this code could cause read-only nodes to attempt to update the accounting.

Reply:
Besides this, it looks ready to merge in.

Reply:
Hm, we have a kinda weird 3-state situation
-no-algod : daemon runs in read-only mode
(nothing specified): daemon isn't read-only, but usually has nothing to do because there's no input
-algod/-algod-net/-algod-token: connects to algod and imports data

I thought this code would only be in the second case, but now I think it would run in the -no-algod case and that would be wrong. I think I'll move it into the reset subcommand and make an option there for `reset -and-rebuild`

Reply:
Ooo, I 'fixed a bug' that exposes that the integration test was relying on the 'nothing specified' behavior above. I changed the code to eliminate that gap so that case behaved like read-only; and the integration tests failed because they rely on running migrations in that state. I think I'll add back an explicit '--allow-migrations' option?

Reply:
I missed this change before, +1 to --allow-migrations.

It means we can also close this ticket: https://github.com/algorand/indexer/issues/187